src/pathfinder.rs

```diff
@@ -0,0 +1,356 @@
+use num::integer::sqrt;
+use pathfinding::prelude::astar;
+use serde::{Deserialize, Serialize};
+use std::borrow::Borrow;
+use std::hash::Hash;
+use std::num::ParseIntError;
+use std::sync::{Arc, RwLock};
+use std::usize::MAX;
+
+// static mut are not thread safe, and manipulating one is only possible behind unsafe tag
+// However, most of the unsafe call should be thread safe because nodes inside this vec are behind a RwLock
```

Comment:
Feel free to correct me on that, I have barely any experience in rust and wasn't able to affirm or infirm that